app/views/govuk_publishing_components/components/_contents_list.html.erb

```diff
@@ -56,7 +59,7 @@
               track_options: {
                 dimension29: contents_item[:text]
               },
-              ga4_link: ga4_link_data
+              ga4_link: ga4_data.to_json
```

Comment:
Quick question: I think the current implementation will always add a `data-ga4-link` attribute. If `ga4_data` is `nil`, then `data-ga4-link` will be present and set to `null` because it's still calling `to_json` on a `nil` value. I've had a quick tinker around and this seems to work as needed: `ga4_link: (ga4_tracking ? ga4_data.to_json : nil)`

What do you think?